### PR TITLE
Add "Don't ask again" option when confirming whether to paste image from the clipboard

### PR DESCRIPTION
**Pull request by ggrelet**
Adding a feature to something that annoyed me: the dialog box asking whether I wanted to paste an image directly from the clipboard.

This use @nathanbuchar's [electron-settings](https://github.com/nathanbuchar/electron-settings).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -207,19 +207,34 @@ app.on('ready', () => {
 
 	const argv = require('minimist')(process.argv.slice(1));
 
+	const settings = require('electron-settings');
```
Comment (sindresorhus) [inline index.js]: Should use https://github.com/sindresorhus/caprine/blob/master/config.js

File: index.js [hunk 2]
```diff
@@ -207,19 +207,34 @@ app.on('ready', () => {
...
+					checkboxLabel: 'Never ask me again',
+					checkboxChecked: false
+				}, (resp, checkboxChecked) => {
+					if (resp === 0) {
+						// User selected send
+						webContents.paste();
+						settings.set('neverAskMeAgain.state', checkboxChecked);
+					}
+				});
+			} else if (settings.get('neverAskMeAgain.state')) {
```
Comment (sindresorhus) [inline index.js]: No point in having a `else if` here with just the inverted version of the original `if`. Just use plain `else`.

File: index.js [hunk 3]
```diff
@@ -207,19 +207,34 @@ app.on('ready', () => {
...
-					webContents.paste();
-				}
-			});
+			if (!settings.get('neverAskMeAgain.state')) {
+				electron.dialog.showMessageBox({
+					type: 'info',
+					buttons: ['Send', 'Cancel'],
+					message: 'Are you sure you want to send the image in the clipboard?',
+					icon: electron.clipboard.readImage(),
+					checkboxLabel: 'Never ask me again',
```
Comment (sindresorhus) [inline index.js]: `Never ask me again` => `Don't ask me again`

File: index.js [hunk 4]
```diff
@@ -207,19 +207,34 @@ app.on('ready', () => {
 
 	const argv = require('minimist')(process.argv.slice(1));
 
+	const settings = require('electron-settings');
+
+	if (!settings.has('neverAskMeAgain')){
```
Comment (sindresorhus) [inline index.js]: The setting key needs a better and more relevant name.

**Comment by sindresorhus** [comment]
Don't commit the package-lock.json file. I'm not interested in having that.

File: index.js (hunk 3 above)
Comment (ggrelet) [inline index.js]: I think both are correct but I changed it as you requested.

Comment (sindresorhus) [inline index.js]: @ggrelet "Don't ask me again" is more common in the system dialog boxes: https://www.google.no/search?q=%22don%27t+ask+me+again%22&source=lnms&tbm=isch&sa=X&ved=0ahUKEwiY2tOqt-jUAhWjQJoKHcZIBWQQ_AUICigB&biw=1440&bih=821

File: index.js (hunk 2 above)
Comment (ggrelet) [inline index.js]: This was marked as an error when launching `$ np minor --no-publish`

File: index.js (hunk 1 above)
Comment (ggrelet) [inline index.js]: Got it

File: index.js (hunk 4 above)
Comment (ggrelet) [inline index.js]: Got it

File: index.js (hunk 3 above)
Comment (ggrelet) [inline index.js]: You're right

**Comment by ggrelet** [comment]
Thank you 👍

**Pull request merged**